### PR TITLE
Fix Case Sensitivity issue for temp Tables 

### PR DESCRIPTION
**Pull request by jinanisha**
#### What type of PR is this?

Bug Fix

#### What does this PR do / why is it needed ?

1/ Temp table column names can be returned in Uppercase instead of default lowercase (depending on Snowflake Account config). This fix checks for quotedIdentifiersIgnoreCase flag in DatabaseConnection and handle the cases in temp table flow accordingly.

2/ Fix Database Snowflake Integration Tests Workflow dependency issue (https://github.com/finos/legend-engine/actions/workflows/database-snowflake-integration-test.yml)

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

No

### CODE REVIEW

File: legend-engine-xts-relationalStore/legend-engine-xt-relationalStore-execution/legend-engine-xt-relationalStore-executionPlan/src/main/java/org/finos/legend/engine/plan/execution/stores/relational/plugin/RelationalExecutionNodeExecutor.java [hunk 1]
```diff
@@ -1981,6 +1985,23 @@ private DoubleStrategyHashMap<Object, Object, SQLExecutionResult> switchedParent
         return parentMap;
     }
 
+    private DoubleStrategyHashMap<Object, Object, SQLExecutionResult> switchedParentHashMapPerChildResult(RelationalGraphObjectsBatch relationalGraphObjectsBatch, int parentIndex, ResultSet childResultSet, Supplier<List<String>> parentPrimaryKeyColumnsSupplier, boolean isDatabaseIdentifiersCaseSensitive)
+    {
+        List<String> parentPrimaryKeyColumnNames = parentPrimaryKeyColumnsSupplier.get();
+        List<Integer> parentPrimaryKeyIndices;
+        if (!isDatabaseIdentifiersCaseSensitive)
+        {
+            parentPrimaryKeyIndices = parentPrimaryKeyColumnNames.stream().map(key -> key.toUpperCase()).map(FunctionHelper.unchecked(childResultSet::findColumn)).collect(Collectors.toList());
```
Comment (abhishoya-gs) [inline legend-engine-xts-relationalStore/legend-engine-xt-relationalStore-execution/legend-engine-xt-relationalStore-executionPlan/src/main/java/org/finos/legend/engine/plan/execution/stores/relational/plugin/RelationalExecutionNodeExecutor.java:1994]: Should it always be toUpperCase? Could we do something like this

```
Map<String,Integer> m = Maps.mutable.empty();
int columnCount = childResultSet.getMetaData().getColumnCount();
for (int i=1; i<=columnCount; i++) {
    m.put(childResultSet.getMetaData().getColumnName(i).toUpperCase(),i);
}
List<Integer> parentPrimaryKeyIndices = parentPrimaryKeyColumnsSupplier.get().stream().map(String::toUpperCase).map(m::get).collect(Collectors.toList());
```

Comment (jinanisha) [reply legend-engine-xts-relationalStore/legend-engine-xt-relationalStore-execution/legend-engine-xt-relationalStore-executionPlan/src/main/java/org/finos/legend/engine/plan/execution/stores/relational/plugin/RelationalExecutionNodeExecutor.java:1994]: yes, it makes more sense.

**Review by epsstan — DISMISSED** [review]

File: legend-engine-xts-relationalStore/legend-engine-xt-relationalStore-execution/legend-engine-xt-relationalStore-executionPlan/src/main/java/org/finos/legend/engine/plan/execution/stores/relational/plugin/RelationalExecutionNodeExecutor.java [hunk 2]
```diff
@@ -2155,6 +2159,20 @@ private static DoubleStrategyHashMap<Object, Object, SQLExecutionResult> switche
         return parentMap;
     }
 
+    private static DoubleStrategyHashMap<Object, Object, SQLExecutionResult> switchedParentHashMapPerChildResult(RelationalGraphObjectsBatch relationalGraphObjectsBatch, int parentIndex, ResultSet childResultSet, Supplier<List<String>> parentPrimaryKeyColumnsSupplier, boolean isDatabaseIdentifiersCaseSensitive)
```
Comment (gs-ssh16) [inline legend-engine-xts-relationalStore/legend-engine-xt-relationalStore-execution/legend-engine-xt-relationalStore-executionPlan/src/main/java/org/finos/legend/engine/plan/execution/stores/relational/plugin/RelationalExecutionNodeExecutor.java:2162]: I dont see the need for a new function like this. Ideally all usages of the function need to be changed. And the new function should be something like

private List<String> upperCaseColumnsIfDbConnectionIsCaseSensistive(List<String>, DatabaseConnection)

**Review by gs-ssh16 — CHANGES_REQUESTED** [review]

File: legend-engine-xts-relationalStore/legend-engine-xt-relationalStore-execution/legend-engine-xt-relationalStore-executionPlan-connection-tests/pom.xml [hunk 3]
```diff
@@ -82,6 +82,7 @@
                                         <include>org.slf4j:slf4j-api:${slf4j.version}</include>
                                         <include>org.slf4j:jcl-over-slf4j:${slf4j.version}</include>
                                         <include>org.slf4j:slf4j-ext:${slf4j.version}</include>
+                                        <include>org.finos.legend.pure:legend-pure-m4:${legend.pure.version}</include>
```
[the remark this answers is not in the document]
Comment (jinanisha) [reply legend-engine-xts-relationalStore/legend-engine-xt-relationalStore-execution/legend-engine-xt-relationalStore-executionPlan-connection-tests/pom.xml]: removed

File: legend-engine-xts-relationalStore/legend-engine-xt-relationalStore-execution/legend-engine-xt-relationalStore-executionPlan/src/main/java/org/finos/legend/engine/plan/execution/stores/relational/plugin/RelationalExecutionNodeExecutor.java (hunk 2 above)
Comment (jinanisha) [reply legend-engine-xts-relationalStore/legend-engine-xt-relationalStore-execution/legend-engine-xt-relationalStore-executionPlan/src/main/java/org/finos/legend/engine/plan/execution/stores/relational/plugin/RelationalExecutionNodeExecutor.java]: done

**Review by gs-ssh16 — APPROVED** [review]

**Pull request merged**